### DISCUSSION TITLE
Get user metadata in action -  Post authentification(internal)

### DISCUSSION THREAD

**Discussion by x610s** [zitadel/zitadel#9093] [Q&A]
![image](https://github.com/user-attachments/assets/0812df1c-29a9-4e16-980c-55b6c2a63bfc)

Hello guys, do you know why i cant see the users metadata using api.metadata?

as you can see my the user Pedro Fermin(and every user) has some metadata, but i cant see on Post authentification(internal)

**Comment by fforootd** [comment]
Hm, of the top of my head I think you should have a `api.metadata` object you can read directly.

At least looking at this code https://github.com/zitadel/actions/blob/main/examples/add_metadata.js#L10

@livio-a am I reading this wrong?

**Reply by x610s** [reply]
i put tried that piece of code (the validation that if api.metadata is undefined stop the function), but as you can imagine it never past that conditional because api.metadata was always undefined :(

**Reply by fforootd** [reply]
Hm, can you share your code you used?
